tensorflow/src/main/scala/com/spotify/featran/tensorflow/package.scala

```diff
@@ -23,9 +23,14 @@ import org.tensorflow.{example => tf}
 package object tensorflow {
 
   case class TensorFlowFeatureBuilder(
-    @transient private val underlying: Features.Builder = tf.Features.newBuilder())
+    @transient private var underlying: Features.Builder = tf.Features.newBuilder())
       extends FeatureBuilder[tf.Example] {
-    override def init(dimension: Int): Unit = underlying.clear()
+    override def init(dimension: Int): Unit = {
+      if (underlying == null) {
```

Comment:
Would this only happen when the underlying object is created on the submitter side and accidentally serialized so we need to recreate it?

Reply:
Yes the instance is almost always created on the driver/`main` side and ser/de-ed into workers.